waspc/cli/Wasp/Cli/Command/Watch.hs

```diff
@@ -80,11 +81,7 @@ watch waspProjectDir outDir = FSN.withManager $ \mgr -> do
     recompile :: IO ()
     recompile = do
       waspSays $ asWaspStartMessage "Recompiling on file change..."
-      compilationResult <- compileIO waspProjectDir outDir
-      case compilationResult of
-        Left err -> waspScreams $ asWaspFailureMessage "Recompilation on file change failed:" ++ err
-        Right () -> waspSays $ asWaspSuccessMessage "Recompilation on file change succeeded."
-      return ()
+      runCommand $ compileAndSetup waspProjectDir outDir
```

Comment:
There is a slight difference now, and that is that messages produced by `compileAndSetup` say `compile` and not `recompile` any more.

That said, I don't think it really matters to the user, so I think that is ok.

Reply:
Yeah, I should have left that in the PR description. But doing it any other way required more IO monad knowledge than I have right now. :)